processing/src/main/java/io/druid/query/select/PagingOffset.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.select;
+
+/**
+ */
+public class PagingOffset
+{
+  private final int offset;
+  private final int threshold;
+  private final boolean descending;
+
+  private int counter;
+
+  public PagingOffset(int offset, int threshold, boolean descending)
+  {
+    this.offset = offset;
+    this.threshold = threshold;
+    this.descending = descending;
+  }
+
+  public boolean isDescending()
+  {
+    return descending;
+  }
+
+  public int startOffset()
+  {
+    return offset;
+  }
+
+  public int startDelta()
+  {
+    return toDelta(offset, descending);
+  }
+
+  public int threshold()
+  {
+    return threshold;
+  }
+
+  public boolean hasNext()
+  {
+    return counter < threshold;
+  }
+
+  public void next()
+  {
+    counter++;
+  }
+
+  public int current()
+  {
+    return offset + (descending ? -counter : counter);
+  }
+
+  static int toDelta(int offset, boolean descending)
```

Comment:
can we have some different names between these offsets and the one defined in the class?